bokehjs/src/lib/models/mappers/continuous_color_mapper.ts

```diff
@@ -33,15 +33,32 @@ export abstract class ContinuousColorMapper extends ColorMapper {
     })
   }
 
+  protected abstract scan<T>(data: Arrayable<number>, palette: Arrayable<T>) : any
```

Comment:
Use `unknown` instead of `any`. The API will be similarly usable, but at least won't turn into a dynamically typed one.

Reply:
Also, please fix whitespace (no space before colon in type position).

Reply:
Alternatively maybe a more specific type could be provided, if there is common ground between types used by more specific classes.

Reply:
Partially addressed in cb2acb9e2. I think the issue about a more specific type will be clearer in the more experimental PR I am going to branch off this one.